render_delegate/render_param.cpp

```diff
@@ -40,6 +41,11 @@ HdArnoldRenderParam::HdArnoldRenderParam()
 {
     _needsRestart.store(false, std::memory_order::memory_order_release);
     _aborted.store(false, std::memory_order::memory_order_release);
+    // If the HDARNOLD_DEBUG_SCENE env variable is defined, we'll want to 
+    // save out the scene every time it's about to be rendered
+    const char *HDARNOLD_DEBUG_SCENE = std::getenv("HDARNOLD_DEBUG_SCENE");
+    if (HDARNOLD_DEBUG_SCENE) 
+        _debugScene = std::string(HDARNOLD_DEBUG_SCENE);
```

Comment:
I am not sure it is safe to construct a string with a nullptr, which would happen if `HDARNOLD_DEBUG_SCENE` is not set

Reply:
if the variable is not set, then the const char* `HDARNOLD_DEBUG_SCENE`would be null, and so we wouldn't contruct a string from it, right ? We have the same code for OCIO config files variables

Reply:
I don't know why I didn't see the condition, all good.